File: samples-java/src/main/java/io/github/ilaborie/loops/samples/anatomy/Accumulate.java
package io.github.ilaborie.loops.samples.anatomy;

import io.github.ilaborie.loops.samples.Element;

import java.util.List;

import static io.github.ilaborie.loops.samples.Elements.getSomeElements;

public class Accumulate {

    public static void main(String[] args) {
        List<Element> elements = getSomeElements();
        int result = accumulate(elements);
        System.out.println(result);
    }

    public static int accumulate(List<Element> elements) {
        int count = 0;

        for (Element elt : elements) {
            int eltSize = elt.getName().length();
            count += eltSize;
        }

        return count;
    }

}

File: samples-kotlin/src/main/kotlin/io/github/ilaborie/loops/samples/anatomy/accumulate.kt
package io.github.ilaborie.loops.samples.anatomy

import io.github.ilaborie.loops.samples.ElementKt
import io.github.ilaborie.loops.samples.getSomeElementsKt


fun main() {
    val elements = getSomeElementsKt()
    val result = accumulate(elements)
    println(result)
}

fun accumulate(elements: List<ElementKt>): Int {
    var count = 0

    for (elt in elements) {
        val eltSize = elt.name.length
        count += eltSize
    }

    return count
}